docs: updates application configuration docs (#2107)
amundsen-io/amundsen: frontend/docs/application_config.md

## Analytics

Amundsen supports pluggable user behavior analytics via the [analytics](https://github.com/DavidWells/analytics) library.

To emit analytics to a given destination, you must use one of the provided plugins (open a PR if you need to install a different vendor), then specify it the config passing the configuration of your account. Multiple destinations are supported if you wish to emit to multiple backends simultaneously.

We provide out of the box support for Mixpanel, Segment and Google Analytics. All [`@analytics/` plugins](https://github.com/DavidWells/analytics#analytic-plugins) are potentially supported, but you must first install the plugin: `npm install @analytics/<provider>` and send us a PR with it before you can use it.

### Examples
For example, to use Google analytics, you must add the import at the top of your `config-custom.ts` file: `import googleAnalytics from '@analytics/google-analytics';`, then add this config block:

```js
analytics: {
  plugins: [
    googleAnalytics({
      trackingId: '<YOUR_UA_CODE>',
      sampleRate: 100
    }),
  ],
}
```

## Announcements

> Annoncements is a feature that allows to disclose new features, changes or any other news to Amundsen's users using a panel in the homepage.

Reviewer: ```suggestion
Announcements is a feature that allows to disclose new features, changes or any other news to Amundsen's users using a panel in the homepage.
```

Refer to [announcement_client.md](https://github.com/amundsen-io/amundsen/blob/main/frontend/docs/examples/announcement_client.md) for information about fetching announcements.

### Examples
To turn announcements on, change the flag as below:
```js
announcements: {
  enabled: true,
},
```

## Badges

Badges are a special type of tag that cannot be edited through the UI.

> They [`BadgeStyleConfig` type](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-types.ts#L273) can be used to customize the text and color of badges. This config defines a mapping of badge name to a [`BadgeStyle`](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-types.ts#L264) and optional `displayName`. Badges that are not defined will default to use the `BadgeStyle.default` style and `displayName` use the badge name with any `_` or `-` characters replaced with a space.

Reviewer: ```suggestion
The [`BadgeStyleConfig` type](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-types.ts#L273) can be used to customize the text and color of badges. This config defines a mapping of badge name to a [`BadgeStyle`](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-types.ts#L264) and optional `displayName`. Badges that are not defined will default to use the `BadgeStyle.default` style and `displayName` use the badge name with any `_` or `-` characters replaced with a space.
```

#### Examples
Here is how you would add curated tags to the top of the Browse page:
```js
browse: {
  curatedTags: ['tag1', 'tag2', 'tag3'],
  //...
},
```

#### Hide Non-Clickable Badges
 The `BrowseConfig.hideNonClickableBadges` hides non-clickable badges in the homepage if `true`. By default is `false`.

#### Examples
This is how you turn it to true:
```js
browse: {
  hideNonClickableBadges: true,
  //...
},
```

#### Show All Tags

The `BrowseConfig.showAllTags` flag allows us to configure whether we should show all the tags or only the curated ones. The default is `true`.

#### Examples
This option shows all tags when true, or only curated tags if false:
```js
browse: {
  showAllTags: false,
  //...
},
```

#### Show Badges in homepage

By default, all available badges are shown on the homepage. The `browse.showBadgesInHome` configuration can be set to `false` to disable this. In addition, it is possible to hide the "non-clickable" badges using `browse.hideNonClickableBadges` configuration.

#### Examples
Here is how you would remove the badges list from the homepage
```js
browse: {
  //...
  showBadgesInHome: false,
},
```

## Column Lineage

> *TODO

Author: @allisonsuarez would love your help for this one, I guess you have it fresh.

```js
date: {
  default: 'MMM DD, YYYY',
  dateTimeShort: 'MMM DD, YYYY ha z',
  dateTimeLong: 'MMMM Do YYYY [at] h:mm:ss a',
}
```

[Read here the reference](https://devhints.io/datetime#momentjs-format) for formatting.

### Examples
Set these values to get different formats:
```js
date: {
  default: 'YYYY-MM-DD',
  dateTimeShort: 'DD. MMM. YYYY hh:mm',
  dateTimeLong: 'DD. MMM. YYYY hh:mm:ss',
}
```

## Document Title
This configuration string specifies the root of the application title. By default this is 'Amundsen - Data Discovery Portal'.

### Examples
You can set this value with your company name:
```js
documentTitle: 'ACME - Amundsen - Data Discovery Portal',
```

## Editable Text

The [`EditableTextConfig`](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-types.ts#L465) configuration object allows us to configure the maximum length limits for editable fields.

With it, we configure the max length for table and column descriptions, with defaults being:
```js
editableText: {
  columnDescLength: 250,
  tableDescLength: 750,
},
```

### Examples

> To chage this values, simply update those character lengths:

Reviewer: ```suggestion
To change these values, simply update the character lengths:
```
